dirtify.py

```diff
@@ -60,41 +62,22 @@
 if '.DS_Store' in images:
     images.remove('.DS_Store') # For macOS: Skip invisible Desktop Services Store file.
 
-for i, img in tqdm(enumerate(images), total=len(images)):
-    filename = f"{raw_dir}/{images[i]}"
-    img = cv2.imread(filename, cv2.IMREAD_COLOR)
-    
-    # resize and crop to make a 64 x 64 image
-    in_h = img.shape[0]
-    in_w = img.shape[1]
-    scale = max(out_h / in_h, out_w / in_w)
-
-    # resize...
-    img = cv2.resize(img, None, fx = scale, fy = scale, interpolation = cv2.INTER_AREA)
-    mid_h = scale * in_h / 2
-    mid_w = scale * in_w / 2
-
-    # ...and crop
-    img = img[round(mid_h - out_h/2):round(mid_h + out_h/2), round(mid_w - out_w/2):round(mid_w + out_w/2)]
-    out_name = os.path.splitext(images[i])[0] + ".png" # Change the extension to png (for lossless image)
-    cv2.imwrite(f"{clean_dir}/{out_name}", img)
-
-    # Add that dirt!
-    dirty = img
-    
+# Dirtify function
+def add_dirt(in_img, w, h):
+
     if args.invert:
         # Invert the image
-        dirty = 255 - dirty
+        img = 255 - in_img
 
     if args.jpeg is not None:
         # Compress with JPEG at given quality, then uncompress
         encode_param = [int(cv2.IMWRITE_JPEG_QUALITY), args.jpeg]
-        result, jpeg_bytes = cv2.imencode('.jpg', dirty, encode_param)
-        dirty = cv2.imdecode(jpeg_bytes, cv2.IMREAD_COLOR)
+        result, jpeg_bytes = cv2.imencode('.jpg', in_img, encode_param)
+        img = cv2.imdecode(jpeg_bytes, cv2.IMREAD_COLOR)
 
     if args.noise is not None:
         # Mix 3 channels of simple noise over the image
-        h, w, d = dirty.shape[0], dirty.shape[1], dirty.shape[2]
+        h, w, d = in_img.shape[0], in_img.shape[1], in_img.shape[2]
         noise_img = np.random.rand(h,w,d) * 255
         mask_img = np.random.rand(h,w,d)
         for y in range(h):
```

Comment:
You can avoid this triply nested loop by using `itertools.product()`.
- https://ttl255.com/pytips-2-use-itertools-product-to-replace-nested-loops/

Reply:
This is so cool. Done.